File: backend/backend/src/main/java/Shapes/RegularPolygon.java
package Shapes;

import HelpingClasses.RegularPolygonSize;
import HelpingClasses.Size;

public abstract class RegularPolygon extends ClosedShape
{
    private RegularPolygonSize mySize;

    @Override
    public void SetSize(Size size)
    {
        this.mySize = (RegularPolygonSize) size;
    }

    @Override
    public Size GetSize()
    {
        return this.mySize;
    }

    public RegularPolygon(int id)
    {
        super(id);
    }


    @Override
    public void Clone(Shape shape)
    {
        super.Clone(shape);
    }
}

File: frontend/paint_frontend/src/app/classes/RegularPolygon.ts
import { ClosedShape } from "./ClosedShape";
import { Position } from "./Position";
import { RegularPolygonSize } from "./RegularPolygonSize";

export abstract class RegularPolygon extends ClosedShape{
    constructor(id:number, size:RegularPolygonSize, strokeWidth?:number,
        enabled?:boolean, strokeColor?:string,
        fillColor?:string, shapePosition?: Position){

            super(id, size, strokeWidth,
                enabled, strokeColor, fillColor, shapePosition);
    }

    override setSize(size:RegularPolygonSize){
        this.size = size;
    }
}
